Release jobs check out a tag that does not exist yet when the release is a draft

The four release jobs check out the released **tag**. With `"draft": true` in `release-please-config.json`, that tag does not exist yet when they run, so all four fail at checkout and no artifacts are ever built.

## What happens

```
[command] git fetch --no-tags --prune --depth=1 origin \
  +refs/heads/butler-sheet-icons-v5.0.0*:refs/remotes/origin/butler-sheet-icons-v5.0.0* \
  +refs/tags/butler-sheet-icons-v5.0.0*:refs/tags/butler-sheet-icons-v5.0.0*
The process '/usr/bin/git' failed with exit code 1
Waiting 14 seconds before trying again
...
##[error]The process '/usr/bin/git' failed with exit code 1
```

`release-linux`, `release-win64`, `release-macos-arm64` and `sbom-build` all fail at the same step, `Checkout the released tag`. Everything upstream — unit tests, all four integration suites, all four docker builds, and `release-please` itself — is green.

## Why the tag is not there

**A draft release does not create a tag ref.** GitHub stores the intended `tag_name` on the release object and only creates `refs/tags/…` when the release is published. Its own output for the run says so:

```
tag_name : butler-sheet-icons-v5.0.0
html_url : https://github.com/ptarmiganlabs/butler-sheet-icons/releases/tag/untagged-3f4214c5673691de9344
```

`untagged-…`. Confirmed independently: `GET /git/ref/tags/butler-sheet-icons-v5.0.0` returns 404, and `git ls-remote --tags` does not list it while every published release's tag is present.

## Why it has not bitten before

`"draft": true` has been set since `release-please-config.json` was created, so drafts are not the change. The checkout is. At `butler-sheet-icons-v4.1.0` the step reads:

```yaml
- name: Checkout repository
  uses: actions/checkout@…
  with:
    persist-credentials: false
```

No `ref:` — it built `github.sha`, which always exists, so publication state never mattered. #1130 changed it to `ref: ${{ needs.release-please.outputs.release_tag_name }}`, and this is the first release cut since.

That change fixed something real and worse: without a `ref`, the jobs built whatever commit started the run, which shipped all three 4.1.0 binaries stamped 4.0.0. The goal was right; the handle was not.

## Fix

`release-please-action` already emits the released commit, and `ci.yaml` already echoes it in the "Show output from Release-Please" step:

```
sha : cba639fd715ade742eaa51b8f75869d604aff532
```

It is simply not exposed as a job output. Two changes:

```yaml
# in the release-please job
outputs:
  release_sha: ${{ steps.release.outputs['sha'] }}
```

```yaml
# in each of the four release jobs
ref: ${{ needs.release-please.outputs.release_sha }}
```

This keeps both properties #1130 wanted — a build pinned to a known commit, immune to the `github.sha` race — and drops the dependency on a ref that draft mode withholds by design. A sha is also the better provenance anchor of the two, since a tag can be moved and a sha cannot.

## Alternatives considered

- **Drop `"draft": true`.** Works, but every release then becomes public before its binaries are attached, which is what the draft is protecting against.
- **Create the tag in the release-please job.** More moving parts, and it duplicates what publishing already does.

## Acceptance criteria

- [ ] `release-please` exposes the released commit as a job output
- [ ] All four release jobs check out that commit rather than the tag name
- [ ] A release cut with `draft: true` builds and uploads artifacts without manual intervention
- [ ] The version stamped in the artifacts matches the release version — the property #1130 was protecting

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -1024,6 +1024,10 @@ jobs:
     outputs:
       releases_created: ${{ steps.release.outputs.releases_created }}
       release_tag_name: ${{ steps.release.outputs['tag_name'] }}
+      # The commit release-please released. Unlike the tag, this exists the moment the
+      # release object does - a draft release carries a tag *name* but no ref, so the
+      # tag cannot be checked out until someone publishes. Issue #1145.
+      release_sha: ${{ steps.release.outputs['sha'] }}
       release_version: ${{ steps.release.outputs['version'] }}
       release_upload_url: ${{ steps.release.outputs['upload_url'] }}
     env:
@@ -1086,9 +1090,12 @@ jobs:
         uses: actions/checkout@3d3c42e5aac5ba805825da76410c181273ba90b1 # v7.0.1
         with:
           # Without this, the default checkout is `github.sha` - the commit that happened to
-          # trigger this run, which is not the commit release-please just tagged. See the note
+          # trigger this run, which is not the commit release-please just released. See the note
           # on the same step in release-macos-arm64 for what that shipped.
-          ref: ${{ needs.release-please.outputs.release_tag_name }}
+          #
+          # The sha rather than the tag: the release is a draft, and a draft has no tag ref
+          # until it is published. Issue #1145.
+          ref: ${{ needs.release-please.outputs.release_sha }}
           fetch-depth: 2
           persist-credentials: false
 
@@ -1190,7 +1197,11 @@ jobs:
           # that shipped all three 4.1.0 binaries built from the commit before the version bump,
           # so they reported "App version: 4.0.0": the SEA build embeds package.json as an asset
           # and globals.js reads the version straight out of it.
-          ref: ${{ needs.release-please.outputs.release_tag_name }}
+          #
+          # The sha rather than the tag, because release-please is configured to cut a *draft*
+          # release and a draft has no tag ref - GitHub creates it at publication. Checking out
+          # the tag name failed all four release jobs on 5.0.0. Issue #1145.
+          ref: ${{ needs.release-please.outputs.release_sha }}
           persist-credentials: false
 
       - name: Setup Node.js
@@ -1293,7 +1304,11 @@ jobs:
           # that shipped all three 4.1.0 binaries built from the commit before the version bump,
           # so they reported "App version: 4.0.0": the SEA build embeds package.json as an asset
           # and globals.js reads the version straight out of it.
-          ref: ${{ needs.release-please.outputs.release_tag_name }}
+          #
+          # The sha rather than the tag, because release-please is configured to cut a *draft*
+          # release and a draft has no tag ref - GitHub creates it at publication. Checking out
+          # the tag name failed all four release jobs on 5.0.0. Issue #1145.
+          ref: ${{ needs.release-please.outputs.release_sha }}
           persist-credentials: false
 
       # Fail before spending ten minutes on a build that cannot be signed.
@@ -1400,7 +1415,11 @@ jobs:
           # that shipped all three 4.1.0 binaries built from the commit before the version bump,
           # so they reported "App version: 4.0.0": the SEA build embeds package.json as an asset
           # and globals.js reads the version straight out of it.
-          ref: ${{ needs.release-please.outputs.release_tag_name }}
+          #
+          # The sha rather than the tag, because release-please is configured to cut a *draft*
+          # release and a draft has no tag ref - GitHub creates it at publication. Checking out
+          # the tag name failed all four release jobs on 5.0.0. Issue #1145.
+          ref: ${{ needs.release-please.outputs.release_sha }}
           persist-credentials: false
 
       - name: Setup Node.js

